woocommerce/woocommerce
# Introduce opt-in WooCommerce rebranding

### Changes proposed in this Pull Request:

This pull request introduces the styling changes required to conform to the WooCommece rebranding to be rolled out in the following months. The changes are hidden by default, to acivate them one of the following snippets is required:

* `define('WOOCOMMERCE_USE_NEW_BRANDING', true;` - for example in `wp-config.php`.
* `add_filter('woocommerce_use_new_branding', '__return_true');` - for example using [a snippets extension](https://wordpress.org/plugins/code-snippets/).

A new `BrandingController` class and a few additional scss files are added, but these are temporary and only in order to be able to switch between the old and new branding; these files will be removed in a later release of WooCommerce.

The implemented changes are:

1. New icon in the admin dashboard:

![image](https://github.com/user-attachments/assets/fdd65dcb-4260-4da3-86ec-da756253c99b)
![image](https://github.com/user-attachments/assets/010a1d84-bfa2-4631-aa50-2a6679104b36)

2. New default color in emails:

![image](https://github.com/user-attachments/assets/bc452e8d-133a-48bf-a338-ee6a490ea33f)
![image](https://github.com/user-attachments/assets/8e5504a8-f3d6-40df-827b-924937f3ddfe)

**NOTE!** What changes is the _default_ value for the email color setting only, the currently configured color won't change automatically (with the changes of this pull request).

3. New primary color in the admin area:

![image](https://github.com/user-attachments/assets/ae10c0de-769d-4be7-a6bf-788211e310e4)
![image](https://github.com/user-attachments/assets/f63533b4-6b49-400f-bccc-51756b96b9e9)

### How to test the changes in this Pull Request:

Prepare a site with WooCommerce and:

- One product with reviews.
- One order, created either via checkout in the shop or directly in the admin area.

Also add the following snippet to the site:

```php
add_action( 'admin_init', function() {
    WC_Admin_Notices::add_notice( 'redirect_download_method' );
} );
```

1. Activate the new branding by either creating a `WOOCOMMERCE_USE_NEW_BRANDING` constant that returns `true`, or having the `woocommerce_use_new_branding` filter return true, as explained above.

2. Load the admin area of the site. Verify that the left menu shows the new icon in the "WooCommerce" entry, both when the menu is collapsed and expanded.

3. Go to Dashboard - Home (the main WordPress dahsboard, not the WooCommerce one). Verify that the "Redirect only" notice that appears at the top and the stars in "WooCommerce Recent Reviews" use the new color, `#720EEC`; you can use the inspector in the browser developer tools for that, e.g. for the notice:

![image](https://github.com/user-attachments/assets/594ad560-353f-459d-b5fa-7e7fba09ba68)

**NOTE!** In other parts of the admin area (e.g. Products) you may see that the notice uses the default WordPress coloring instead. That's because a `--wp-admin-theme-color` CSS variable exists, but if you use the browser inspector you should see that the fallback color to use when that variable doesn't exist is the new one:

![image](https://github.com/user-attachments/assets/b19152c2-a951-425f-bbac-10ad18f8c6d0)

4. Go to WooCommerce- Settings - Email. Scroll down to the general settings for emails. You will see that the value for "Base color" hasn't changed, but if you click on the "?" icon the tooltip will show the new color as the default for the setting.

Here you can manually change the value to the new default (`#720EEC`), save, and send a test email (e.g. open the details of an order in the admin area, then "Order actions" - "Resend new order notification"). You can use [an email logger extension](https://wordpress.org/plugins/email-log/) if you don't want to send actual emails.

An alternative way to verify that the new default is actually a default could be:

* Install WordPress without WooCommerce.
* Add the constant or the filter to activate the new branding.
* Install WooCommerce from a zip file generated from this branch.
* Verify that the value of the base color setting for emails is the new default value.

5. Repeat the previous steps without the constant or filter, verify that the old branding is restored (the old default primary color for the admin area and emails is `#7F54B3`).

### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
